opslevel/resource_opslevel_user.go

```diff
@@ -1,123 +1,189 @@
-// 	err := client.DeleteUser(id)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	d.SetId("")
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &UserResource{}
+
+var _ resource.ResourceWithImportState = &UserResource{}
+
+func NewUserResource() resource.Resource {
+	return &UserResource{}
+}
+
+// UserResource defines the resource implementation.
+type UserResource struct {
+	CommonResourceClient
+}
+
+// UserResourceModel describes the User managed resource.
+type UserResourceModel struct {
+	Email            types.String `tfsdk:"email"`
+	Id               types.String `tfsdk:"id"`
+	LastUpdated      types.String `tfsdk:"last_updated"`
+	Name             types.String `tfsdk:"name"`
+	Role             types.String `tfsdk:"role"`
+	SkipWelcomeEmail types.Bool   `tfsdk:"skip_welcome_email"` // not usable but kept for backwards compatibility
+}
+
+func NewUserResourceModel(user opslevel.User) UserResourceModel {
+	return UserResourceModel{
+		Email:            types.StringValue(user.Email),
+		Id:               types.StringValue(string(user.Id)),
+		LastUpdated:      types.StringValue(timeLastUpdated()),
+		Name:             types.StringValue(user.Name),
+		Role:             types.StringValue(string(user.Role)),
+		SkipWelcomeEmail: types.BoolNull(),
+	}
+}
+
+func (r *UserResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_user"
+}
+
+func (r *UserResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "User Resource",
+
+		Attributes: map[string]schema.Attribute{
+			"email": schema.StringAttribute{
+				Description: "The email address of the user.",
+				Required:    true,
+			},
+			"id": schema.StringAttribute{
+				Description: "The ID of the user.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"last_updated": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the user.",
+				Required:    true,
```

Comment:
```suggestion
				Optional:    true,
```

Reply:
The `name` field is required in the current provider - [docs](https://registry.terraform.io/providers/OpsLevel/opslevel/latest/docs/resources/user)